suite-common/fiat-services/src/rates.ts

```diff
@@ -1,35 +1,128 @@
-import type { TickerId, TimestampedRates, LastWeekRates } from '@suite-common/wallet-types';
+import { getUnixTime, subWeeks } from 'date-fns';
+
+import type { TickerId, LastWeekRates } from '@suite-common/wallet-types';
 import { FiatCurrencyCode } from '@suite-common/suite-config';
+import TrezorConnect from '@trezor/connect';
+import { scheduleAction } from '@trezor/utils';
 
 import * as coingeckoService from './coingecko';
 import * as blockbookService from './blockbook';
 
 export const { getTickerConfig, fetchCurrentTokenFiatRates } = coingeckoService;
 
-export const fetchCurrentFiatRates = async (ticker: TickerId): Promise<TimestampedRates | null> => {
-    const res = blockbookService.isTickerSupported(ticker)
-        ? await blockbookService.fetchCurrentFiatRates(ticker.symbol)
+const CONNECT_FETCH_TIMEOUT = 10_000;
+
+const getConnectFiatRatesForTimestamp = (
+    ticker: TickerId,
+    timestamps: number[],
+    currency: FiatCurrencyCode,
+) =>
+    scheduleAction(
+        () =>
+            TrezorConnect.blockchainGetFiatRatesForTimestamps({
+                coin: ticker.symbol,
+                token: ticker.tokenAddress,
+                timestamps,
+                currencies: [currency],
+            }),
+        {
+            timeout: CONNECT_FETCH_TIMEOUT,
+        },
+    );
+
+export const fetchCurrentFiatRates = async (
+    ticker: TickerId,
+    currency: FiatCurrencyCode,
+): Promise<number | null | undefined> => {
+    const responseConnect = await scheduleAction(
+        () =>
+            TrezorConnect.blockchainGetCurrentFiatRates({
+                coin: ticker.symbol,
+                token: ticker.tokenAddress,
+                currencies: [currency],
+            }),
+        { timeout: CONNECT_FETCH_TIMEOUT },
+    );
+
+    const rateFromConnect = responseConnect.success
+        ? responseConnect.payload.rates?.[currency]
         : null;
-    return res ?? coingeckoService.fetchCurrentFiatRates(ticker);
+
+    if (rateFromConnect) {
+        return rateFromConnect;
+    }
+
+    if (ticker.tokenAddress) {
+        // We don't want to fallback to coingecko for tokens because it returns nonsenses
```

Comment:
This means that we have no way of fetching fiat rates for ADA and SOL tokens, right?

Reply:
True. In my opinion this could be deleted at all, we are already using Coingecko for ETH tokens under the hood in TrezorConnect methods since it gets values from blockbook.

Reply:
Yes, added fetching for them [364a043](https://github.com/trezor/trezor-suite/pull/10385/commits/364a04314734bcd1f9afd621898bdae6e9064282)

Reply:
<img width="821" alt="Screenshot 2024-01-02 at 4 43 16" src="https://github.com/trezor/trezor-suite/assets/33235762/fd90bd7e-6a49-4dbc-b136-aceb576f8e6b">